tools/build-tools/src/typeValidator/testGeneration.ts

```diff
@@ -16,27 +16,36 @@ export function generateTests(packageDetails: PackageDetails, packageDir: string
 `/*!
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
+ *
+ * THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+ * It is created by fluid-type-validator in @fluidframework/build-tools
  */
 
 import * as old from "${oldVersion}";
 import * as current from "../index";
 `
             ];
         const oldDetails = generateTypeDataForProject(packageDir, oldVersion);
-        const oldTypes = oldDetails.typeData;
-        for(const type of currentTypeData.typeData){
-            const typeString = type.name.replace(".","");
+        const currentTypeMap = new Map<string, TypeData>(currentTypeData.typeData.map((v)=>[getFullTypeName(v),v]));
+        for(const oldTypeData of oldDetails.typeData){
             // no need to test new types
-            if(oldTypes.some((t)=>t.name.replace(".","") == typeString)){
-                const oldType = `old.${type.name}`
-                const currentType = `current.${type.name}`
+            if(currentTypeMap.has(getFullTypeName(oldTypeData))){
+                const oldType: TestCaseTypeData = {
+                    prefix: "old",
+                    ... oldTypeData,
+                }
+                const currentType: TestCaseTypeData = {
+                    prefix: "current",
+                    ... currentTypeMap.get(getFullTypeName(oldTypeData))!,
+
+                }
 
                 testString.push(`/*`)
                 testString.push(`* validate forward compat by using old type in place of current type`);
                 testString.push(`* to disable, add in package.json under typeValidation.broken:`);
-                testString.push(`* "${type.name}": {"forwardCompat": false}`);
+                testString.push(`* "${getFullTypeName(currentType)}": {"forwardCompat": false}`);
```

Comment:
nit: could refactor this

Reply:
not quite following. what's the suggested refactor here?